Warnings when building C code

Currently the C code in here for the different impls (my_full_neuron_impl.h and my_neuron_impl_semd.h) builds fine, but with warnings, e.g.

> 
> In file included from <command-line>:
> /Users/andrewgait/work/spinnaker/git/sPyNNaker8NewModelTemplate/c_models/modified_src/my_models/implementations/my_full_neuron_impl.h:66:13: warning: ‘neuron_impl_do_timestep_update’ defined but not used [-Wunused-function]
>    66 | static bool neuron_impl_do_timestep_update(
>       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /Users/andrewgait/work/spinnaker/git/sPyNNaker8NewModelTemplate/c_models/modified_src/my_models/implementations/my_full_neuron_impl.h:56:13: warning: ‘neuron_impl_add_inputs’ defined but not used [-Wunused-function]
>    56 | static void neuron_impl_add_inputs(
>       |             ^~~~~~~~~~~~~~~~~~~~~~
> /Users/andrewgait/work/spinnaker/git/sPyNNaker8NewModelTemplate/c_models/modified_src/my_models/implementations/my_full_neuron_impl.h:49:13: warning: ‘neuron_impl_store_neuron_parameters’ defined but not used [-Wunused-function]
>    49 | static void neuron_impl_store_neuron_parameters(
>       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /Users/andrewgait/work/spinnaker/git/sPyNNaker8NewModelTemplate/c_models/modified_src/my_models/implementations/my_full_neuron_impl.h:42:13: warning: ‘neuron_impl_load_neuron_parameters’ defined but not used [-Wunused-function]
>    42 | static void neuron_impl_load_neuron_parameters(
>       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /Users/andrewgait/work/spinnaker/git/sPyNNaker8NewModelTemplate/c_models/modified_src/my_models/implementations/my_full_neuron_impl.h:29:13: warning: ‘neuron_impl_initialise’ defined but not used [-Wunused-function]
>    29 | static bool neuron_impl_initialise(uint32_t n_neurons) {

I think this probably ought to be fixed by using .c files rather than .h files for impls, but open for suggestions.

--- a/c_models/src/my_models/implementations/my_full_neuron_impl.h
+++ b/c_models/src/my_models/implementations/my_full_neuron_impl.h
@@ -26,6 +26,7 @@ typedef struct neuron_impl_t {
 //! Array of neuron states
 static neuron_impl_t *neuron_array;
 
+__attribute__((unused)) // Marked unused as only used sometimes
 static bool neuron_impl_initialise(uint32_t n_neurons) {
     // Allocate DTCM for neuron array
     if (sizeof(neuron_impl_t) != 0) {
@@ -39,6 +40,7 @@ static bool neuron_impl_initialise(uint32_t n_neurons) {
     return true;
 }
 
+__attribute__((unused)) // Marked unused as only used sometimes
 static void neuron_impl_load_neuron_parameters(
         address_t address, uint32_t next, uint32_t n_neurons) {
     // Copy parameters to DTCM from SDRAM
@@ -46,6 +48,7 @@ static void neuron_impl_load_neuron_parameters(
             n_neurons * sizeof(neuron_impl_t));
 }
 
+__attribute__((unused)) // Marked unused as only used sometimes
 static void neuron_impl_store_neuron_parameters(
         address_t address, uint32_t next, uint32_t n_neurons) {
     // Copy parameters to SDRAM from DTCM
@@ -53,6 +56,7 @@ static void neuron_impl_store_neuron_parameters(
             n_neurons * sizeof(neuron_impl_t));
 }
 
+__attribute__((unused)) // Marked unused as only used sometimes
 static void neuron_impl_add_inputs(
         index_t synapse_type_index, index_t neuron_index,
         input_t weights_this_timestep) {
@@ -63,6 +67,7 @@ static void neuron_impl_add_inputs(
     neuron->inputs[synapse_type_index] += weights_this_timestep;
 }
 
+__attribute__((unused)) // Marked unused as only used sometimes
 static bool neuron_impl_do_timestep_update(
         index_t neuron_index, input_t external_bias) {
     // Get the neuron itself

--- a/c_models/src/my_models/implementations/my_neuron_impl_semd.h
+++ b/c_models/src/my_models/implementations/my_neuron_impl_semd.h
@@ -42,6 +42,7 @@ static threshold_type_pointer_t threshold_type_array;
 // The synapse shaping parameters
 static synapse_param_t *neuron_synapse_shaping_params;
 
+__attribute__((unused)) // Marked unused as only used sometimes
 static bool neuron_impl_initialise(uint32_t n_neurons) {
     // Allocate DTCM for neuron array
     neuron_array = spin1_malloc(n_neurons * sizeof(neuron_t));
@@ -77,6 +78,7 @@ static bool neuron_impl_initialise(uint32_t n_neurons) {
     return true;
 }
 
+__attribute__((unused)) // Marked unused as only used sometimes
 static void neuron_impl_add_inputs(
         index_t synapse_type_index, index_t neuron_index,
         input_t weights_this_timestep) {
@@ -86,6 +88,7 @@ static void neuron_impl_add_inputs(
             parameters, weights_this_timestep);
 }
 
+__attribute__((unused)) // Marked unused as only used sometimes
 static void neuron_impl_load_neuron_parameters(
         address_t address, uint32_t next, uint32_t n_neurons) {
     log_debug("writing parameters, next is %u, n_neurons is %u ",
@@ -110,6 +113,7 @@ static void neuron_impl_load_neuron_parameters(
             n_neurons * sizeof(synapse_param_t));
 }
 
+__attribute__((unused)) // Marked unused as only used sometimes
 static bool neuron_impl_do_timestep_update(index_t neuron_index,
         input_t external_bias) {
     // Get the neuron itself
@@ -193,6 +197,7 @@ static bool neuron_impl_do_timestep_update(index_t neuron_index,
 
 //! \brief stores neuron parameter back into sdram
 //! \param[in] address: the address in sdram to start the store
+__attribute__((unused)) // Marked unused as only used sometimes
 static void neuron_impl_store_neuron_parameters(
         address_t address, uint32_t next, uint32_t n_neurons) {
     log_debug("writing parameters");

